spring-cloud-stream-core-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc

```diff
@@ -678,6 +655,202 @@ boolean result = pollableSource.poll(received -> {
+headers:
+x-original-exchange:
+x-exception-message:	has an error
+x-original-routingKey:	input.myGroup
+x-exception-stacktrace:	org.springframework.messaging.MessageHandlingException: nested exception is
+      org.springframework.messaging.MessagingException: has an error, failedMessage=GenericMessage [payload=byte[15],
+      headers={amqp_receivedDeliveryMode=NON_PERSISTENT, amqp_receivedRoutingKey=input.hello, amqp_deliveryTag=1,
+      deliveryAttempt=3, amqp_consumerQueue=input.hello, amqp_redelivered=false, id=a15231e6-3f80-677b-5ad7-d4b1e61e486e,
+      amqp_consumerTag=amq.ctag-skBFapilvtZhDsn0k3ZmQg, contentType=application/json, timestamp=1522327846136}]
+      at org.spring...integ...han...MethodInvokingMessageProcessor.processMessage(MethodInvokingMessageProcessor.java:107)
+      at. . . . .
+Payload {"name”:"Bob"}
+----
+
+====== Re-queue
+
+As mentioned earlier, the currently supported binders (Rabbit and Kafka) rely on `RetryTemplate` to facilitate successful message processing. See <<Retry Template>> for details.
+However, for cases when `max-attempts` property is set to 1, internal reprocessing of the message is disabled, at this point you can facilitate message re-processing (re-tries)
+by instructing messaging system to simply re-queue the failed message. Once re-queued the failed message will be re-sent back to the original handler, essentially creating a retry loop.
+
+This option may be feasible for cases where the nature of the error is related to some sporadic yet short term unavailability of some resource.
+
+To accomplish that you must set the following properties:
+
+[source,text]
+----
+spring.cloud.stream.bindings.input.consumer.max-attempts=1
+spring.cloud.stream.rabbit.bindings.input.consumer.requeue-rejected=true
+----
+
+In the above, the `max-attempts` set to 1 essentially disabling internal re-tries and `requeue-rejected` (short for _requeue rejected messages_) os set to `true`.
+Once set the failed message will be resubmitted to the same handler and loop continuously or until the handler throws `AmqpRejectAndDontRequeueException`
+essentially allowing you to build your own re-try logic within the handler itself.
+
+===== Retry Template
+
+The `RetryTemplate` is part of the https://github.com/spring-projects/spring-retry[Spring Retry] library.
+While it is out of scope of this dcument to cover all of the capabilities of the `RetryTemplate`, we will mentione the following consumer properties which are specifically related to
+the `RetryTemplate`.
+
+maxAttempts::
+The number of attempts to process the message.
++
+Default: 3.
+backOffInitialInterval::
+The backoff initial interval on retry.
++
+Default 1000 milliseconds.
+backOffMaxInterval::
+The maximum backoff interval.
++
+Default 10000 milliseconds.
+backOffMultiplier::
+The backoff multiplier.
++
+Default 2.0.
+
+While the above is sufficient for majority of the customization requirements, it may not satisfy certain complex requirements at which
```

Comment:
Replace with "While the preceding settings are sufficient for the majority of the customization requirements, they  may not satisfy certain complex requirements, at which"